mlflow/models/evaluation/base.py

```diff
@@ -778,6 +798,70 @@ def evaluate(
                              If multiple evaluators are specified, each configuration should be
                              supplied as a nested dictionary whose key is the evaluator name.
 
+    :param custom_metrics: (Optional) A list of custom metric functions. A custom metric
+                              function is required to take in two parameters:
+                              - Union[pandas.Dataframe, pyspark.sql.DataFrame]: The first being a
+                                Pandas or Spark DataFrame containing ``prediction`` and ``target``
+                                column. The ``prediction`` column contains the predictions made by
+                                the model. The ``target`` column contains the corresponding labels
+                                to the predictions made on that row.
+                              - Dict: The second is a dictionary containing the metrics calculated
+                                by the default evaluator. The keys are the names of the metrics
+                                and the values are the scalar values of the metrics. Refer to the
+                                DefaultEvaluator behavior section for what metrics will be returned
+                                based on the type of model (i.e. classifier or regressor).
+
+                              A custom metric function can return in two different formats:
+                              - Dict[AnyStr, Union[int, float, np.number]: a singular dictionary of
+                                custom metrics, where the keys are the names of the metrics, and the
+                                values are the scalar values of the metrics.
+                              - Tuple[Dict[AnyStr, Union[int,float,np.number]], Dict[AnyStr,Any]]:
+                                a tuple of a dict containing the custom metrics, and a dict of
+                                artifacts, where the keys are the names of the artifacts, and the
+                                values are objects representing the artifacts.
+
+                              Object types that artifacts be represented as:
+                              - A string uri representing the file path to the artifact. MLflow will
+                                infer the type of the artifact based on the file extension.
+                              - ``EvaluationArtifact`` type objects. i.e. CsvEvaluationArtifact,
+                                ImageEvaluationArtifact.
+                              - Pandas DataFrame. This will be resolved as a CSV artifact.
+                              - Numpy array. This will be saved as a .npy artifact.
+                              - Matplotlib Figure. This will be saved as an image artifact.
+                              - Any JSON serializable object. This will be saved as a JSON artifact.
```

Comment:
We have a branch cut for MLflow 1.24.0 coming up within the next 1-2 days. In case we want to include support for numerical metrics in 1.24.0 and follow up with artifacts in 1.25.0, can we remove this portion of the docstring since artifacts aren't yet supported?

Reply:
I think I should also modify the example to exclude artifacts support.

Reply:
👍